physics/cires_ugwp_triggers_v1.F90

```diff
@@ -0,0 +1,576 @@
+module cires_ugwp_triggers_v1
+
+
+contains
+
+
+        subroutine ugwp_triggers
+        implicit none
+        write(6,*) ' physics-based triggers for UGWP ' 
+        end subroutine ugwp_triggers
+!	
+     SUBROUTINE  subs_diag_geo(nx, ny,  lat, lon, rlat, rlon, dy, dx,  &
+                               cosv, rlatc, brcos, brcos2, dlam1, dlam2, dlat, divJp, divJm)
+      use ugwp_common_v1 , only : deg_to_rad
+      
+      implicit none
+       integer :: nx, ny
+       real    :: lon(nx), lat(ny)
+       real    :: rlon(nx), rlat(ny) , cosv(ny), tanlat(ny)  
+       real    :: rlatc(ny-1), brcos(ny), brcos2(ny)              
+       real    :: earth_r, ra1, ra2, dx, dy, dlat
+       real    :: dlam1(ny), dlam2(ny),  divJp(ny), divJm(ny)
+       integer :: j
+!
+!    specify common constants and
+!    geometric factors to compute deriv-es etc ...
+!    coriolis coslat tan etc...
+!
+      earth_r = 6370.e3
```

Comment:
radius of the earth should be defined by the host model (any host model). You want to be consistent, i.e. get it from the host model via the argument list.

Reply:
Done.